packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -1870,6 +1870,14 @@ describe('<Autocomplete />', () => {
       expect(handleChange.callCount).to.equal(1);
       expect(handleChange.args[0][1]).to.equal('あ');
     });
+
+    it('should only render endAdornment when hasClearIcon or hasPopupIcon is true', () => {
+      const { container } = render(
+        <Autocomplete freeSolo options={[]} renderInput={(params) => <TextField {...params} />} />,
+      );
+
+      expect(container.querySelector(`.${classes.root}`)).not.to.have.class(classes.endAdornment);
```

Comment:
```suggestion
expect(container.querySelector(`.${classes.endAdornment}`)).to.equal(null);
```
If you inspect, it's a separate HTML div element and not applied as a class in Autocomplete root div.